src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -111,13 +111,17 @@ internal JsonClassInfo(Type type, JsonSerializerOptions options)
                     }
                 }
             }
-            else if (ClassType == ClassType.Enumerable)
+            else if (ClassType == ClassType.Enumerable || ClassType == ClassType.Dictionary)
             {
                 // Add a single property that maps to the class type so we can have policies applied.
-                AddProperty(type, propertyInfo : null, type, options);
+                JsonPropertyInfo jsonPropertyInfo = AddProperty(type, propertyInfo: null, type, options);
```

Comment:
It's odd to see the same parameter being passed in twice (in `AddProperty`). It looks like 2/3 calls to `AddProperty` has this behavior. Should this be re-structured to avoid that?

Reply:
I'll add a `AddPolicyProperty` with just two args. Thanks